samples/sensor/qdec/boards/mimxrt1050_evk.overlay

```diff
@@ -0,0 +1,52 @@
+/*
```

Comment:
I would refactor this commit to only include the addition of emulation support in the `qdec` sensor sample. Then, take the commit adding pin control support for `qdec `to the RT1050 EVK, and make that your final commit. You can include the overlay in that commit.

So to summarize, a series of commits like the following:
- add QDEC driver
- add DTS and Kconfig changes for RT1050
- add QDEC emulation support to sample
- add RT1050 pinmux and overlay to support QDEC sample

I'm aware this recommendation doesn't align with my prior request, I did not realize the QDEC sample already existed in tree on my first pass of this PR. If the QDEC sample is already upstream, I'd like to keep changes to the sample source code in a separate commit.

Reply:
@danieldegrasse No problem. I refactored the commits :+1: